### PR TITLE
Basic Packagedcode module for handling Opam Package Manger

### LINKED ISSUE #1974: add .opam file checker
## Short Description
Currently there is no implementation for checking .opam file. Opam is a source-based package manager for OCaml. It supports multiple simultaneous compiler installations, flexible package constraints, and a Git-friendly development workflow. opam files contains the information of ocaml project.

## Possible Labels

- new feature

## Select Category

-  Enhancement [ ]
-  Add License/Copyright [ ]
-  Scan Feature [ ]
-  Packaging [x]
-  Documentation [ ]
-  Expand Support [ ]
-  Other [ ]

### PR DESCRIPTION
**Pull request by rpotter12**
Fixes #1974 

**IMPORTANT**
I created pypi package named [opam](https://pypi.org/project/opam/). Through which we can extract data of .opam files. I have created that package in few hours so it is having very limited functions only. To extract the data of .opam file properly we need to implement codes in that properly. 

GitHub link of opam package is: https://github.com/rpotter12/opam

Scancode-toolkit contributors, please contribute in the repository so that we can extract .opam file data properly.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

**Comment by rpotter12** [comment]
For those looking to quickly test this out:

```
$ wget https://raw.githubusercontent.com/ocaml/ocaml/trunk/ocaml-variants.opam`
$ scancode --package ocaml-variants.opam --json-pp -
```

**Comment by rpotter12** [comment]
@pombredanne Please review this PR. :)

**Comment by rpotter12** [comment]
@pombredanne I am not understanding why the tests are failing. Please tell me about it.

**Comment by ritiek** [comment]
@rpotter12 There seems to be one test failing at the moment. If you take a look at the diff of the failing test. You'll see these addtional lines
https://travis-ci.org/github/nexB/scancode-toolkit/jobs/662339226#L4663-L4668
```diff
  Package: nuget
  class: packagedcode.nuget:NugetPackage
  metafiles: [Content_Types].xml, *.nuspec
  extensions: .nupkg
  filetypes: zip archive, microsoft ooxml

  ---------------------------------------------
> Package: opam
>   class: packagedcode.ocaml:OpamPackageManager
>   metafiles: *.opam
>   extensions: .opam
>
> --------------------------------------------
  Package: pypi
    class: packagedcode.pypi:PythonPackage
    metafiles: metadata.json, *setup.py, PKG-INFO, *.whl, *.egg
    extensions: .egg, .whl, .pyz, .pex
```
are expected which are currently missing from this file:
https://github.com/nexB/scancode-toolkit/blob/54f5f689b9be958f3bd871536a5ce3d34a115255/tests/packagedcode/data/plugin/help.txt#L197-L208

Adding those additional lines to this file should fix this.

You can run the currently failing test locally with:
```shell
$ pytest tests/packagedcode/test_plugin.py
```

**Comment by rpotter12** [comment]
@ritiek Thanks buddy. :)

**Comment by rpotter12** [comment]
@pombredanne @majurg Please review :)

File: src/packagedcode/__init__.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 from packagedcode import bower
 from packagedcode import conda
 from packagedcode import cargo
+from packagedcode import ocaml
```
Comment (TG1999) [inline src/packagedcode/__init__.py]: Please sort imports

Comment (TG1999) [reply src/packagedcode/__init__.py]: It should come after `from packagecode import nugget` IMHO

**Comment by rpotter12** [comment]
@pombredanne @majurg Please review this PR. :)

File: src/packagedcode/ocaml.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
+
+# Copyright (c) 2019 nexB Inc. and others. All rights reserved.
```
Comment (majurg) [inline src/packagedcode/ocaml.py]: Lets remove copyright years here

**Comment by rpotter12** [comment]
Hi @majurg 
> 1. is there a reason we need to use your separate python package for this? With the limited amount of code needed here, why can we not just include it in scancode-toolkit?

No, there is no need to use a separate python package for now. We can include the code in scancode-toolkit.
> 2. Where are the tests for this?

I have created only the basic packagedcode module in this PR. If test are needed in this PR, I will include them by today. :)

**Comment by rpotter12** [comment]
@pombredanne @majurg @ritiek Please review the PR again. I have made the required changes but some checks are failing and I don't know why they are. If they are failing because of my codes then please guide me. :)

**Comment by rpotter12** [comment]
@pombredanne @majurg @ritiek Please review the PR again. I have tried by including the code in the scripts but it was failing all the checks by that implementation. I think it will be better to use separate python package for this. Plus if some other file which is having the same code pattern as opam file is having, then it will be helpful. 

Some checks are canceled, I don't know why they canceled. Please review this:)

**Comment by rpotter12** [comment]
@pombredanne @majurg @ritiek Please review the PR again. I have tried by including the code in the scripts but it was failing all the checks by that implementation. I think it will be better to use separate python package for this. Plus if some other file which is having the same code pattern as opam file is having, then it will be helpful.

Some checks are canceled, I don't know why they canceled. Please review this:)

**Comment by rpotter12** [comment]
@majurg I have tried by including the code in the scripts but it was failing all the checks by that implementation. I think it will be better to use a separate python package for now. With further implementation, we will include package code in Scancode.

**Comment by rpotter12** [comment]
@pombredanne @majurg @ritiek Please review the PR again.

**Comment by rpotter12** [comment]
@majurg required changes are done in #1981

**Pull request closed without merging**